### ISSUE TITLE
Less intuitive sharing after updating

### ISSUE THREAD

**Issue by bartekj001** [mozilla-mobile/fenix#14123]
Hi. I update Firefox to version 79.0.5 on Android.

Before the update, when I shared any page from any browser, I had the option to "share to device" without having to launch the page on Firefox. It helped a lot, flipping the pages to the computer for reading later. Now, this option is unavailable. 

Now, in the share menu I only have a "Firefox" icon and the page opens in my browser where I need to find the share icon and then find my device. It takes much more time and is less intuitive.

The second thing, previously, the choice of the device was easier to read because the entire name was displayed. Now there is only the beginning part of the name.

I made a screen of what it looked like before (white theme) and now (black theme).
Best regards.

![fasdfasdfasdfy](https://user-images.githubusercontent.com/70169047/91084008-9b42bf80-e64b-11ea-86a9-a9ee5f74d8e2.jpg)

### Device information

* Android device: Samsung Note 10+
* Fenix version: 79.0.5

**Comment by hwinnemoe** [comment]
Duplicate of #1523

**Comment by jtiemer** [comment]
As issue #1523 was closed for being off-topic, could we reopen this one? The issue in itself is apparently relevant to many users. I myself used the functionality in Fennec a lot as was seriously disappointed when it was removed.

The actual use cases for sharing from an app to the on-device browser elude me while the use case for sharing with my desktop firefox is obvious.

**Comment by iridescentFUZZ** [comment]
> 
> 
> As issue #1523 was closed for being off-topic, could we reopen this one? The issue in itself is apparently relevant to many users. I myself used the functionality in Fennec a lot as was seriously disappointed when it was removed.
> 
> The actual use cases for sharing from an app to the on-device browser elude me while the use case for sharing with my desktop firefox is obvious.

Yeah, I feel the same way. The old share to device function was far more useful.

**Comment by jtiemer** [comment]
Using the new functionality for two months i can underline that the majority of the functionality's usage is sharing from an app (news mostly) to the browser, and from there to my firefox synced device.

Maybe a change of perspective is what is needed to improve this sharing feature. It is, in my opinion, less a feature of the on-device browser, but a feature of Firefox-Sync. With other words: It's most important task is not to connect the on-device firefox with other devices' synced firefox instances, but to allow for the influx of shared URLs/items from any source. One of which is (of course) the on-device firefox instance.

My usecase, as hinted at above is that I share to a different device when I cannot finish a news article since it's too long or if I want to read it again, or if the mobile/app view is reduced in a way that makes me want to view the whole thing in a proper browser.

**Comment by thecosmicfrog** [comment]
Just to chime in, my use case is exactly that of @jtiemer. I like to fire long articles or useful links from my phone to either my work or personal laptops. I don't consider this a feature of the browser, but more so a Firefox Sync and Android sharing feature.
